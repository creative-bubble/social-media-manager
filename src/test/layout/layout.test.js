import { render } from '../helper/helper';
import { fireEvent } from '@testing-library/react';
import Navbar from '../../layout/navbar/index';
import Contact from '../../layout/contact/index';


describe('layouts', () => {
    describe('Navbar', () => {
        it('Render Navbar', () => {
            const { asFragment } = render(Navbar);
            //assertions
            expect(asFragment()).toMatchSnapshot('Navbar-layout');
        });
    
        it('Navbar User button', () => {
            const { getByTestId, getByText, asFragment } = render(Navbar);
            fireEvent.click(getByTestId('user-button'));
            //assertions
            getByText('Profile');
            expect(asFragment()).toMatchSnapshot('Navbar-layout-click-user-button');
        });
    });
    
    describe('Contact', () => {
        it('Render Contact', () => {
            const { asFragment } = render(Contact);
            //assertions
            expect(asFragment()).toMatchSnapshot('Contact-layout');
        }); 
    });

    describe('Footer', () => {
        it('Render Footer', () => {
            const { asFragment } = render(Contact);
            //assertions
            expect(asFragment()).toMatchSnapshot('Footer-layout');
        }); 
    });
});