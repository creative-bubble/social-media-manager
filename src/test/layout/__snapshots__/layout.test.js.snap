// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts Contact Render Contact: Contact-layout 1`] = `
<DocumentFragment>
  <div
    class="container-xl mt-6 pb-20 sm:mt-0"
  >
    <div
      class="relative sm:w-9/12 mx-auto flex flex-col md:flex-row"
    >
      <br
        class="sm:hidden"
      />
      <div
        class=" w-full md:w-6/12"
      >
        <div
          class="w-9/12 mx-auto"
        >
          <div
            class="relative h-16 sm:h-20 flex justify-center sm:justify-start"
          >
            <h2
              class="absolute mt-6 font-semibold text-blue-500"
            >
              CONTACT US
            </h2>
          </div>
          <p
            class="text-xs font-semibold text-blue-500"
          >
            Drop a message
          </p>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <textarea
            class="w-full h-40 mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            placeholder="Message*"
          />
          <button
            class="w-full h-8 mt-4 text-xs rounded-full bg-blue-500 text-gray-200 tracking-wider focus:outline-none active:bg-blue-400 "
          >
            SUBMIT
          </button>
        </div>
      </div>
      <div
        class="absolute bottom-0 right-0 w-6/12 h-64 mx-auto hidden md:block"
      >
        <img
          alt="media1"
          class="h-full w-full object-contain"
          src="[object Object]"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`layouts Footer Render Footer: Footer-layout 1`] = `
<DocumentFragment>
  <div
    class="container-xl mt-6 pb-20 sm:mt-0"
  >
    <div
      class="relative sm:w-9/12 mx-auto flex flex-col md:flex-row"
    >
      <br
        class="sm:hidden"
      />
      <div
        class=" w-full md:w-6/12"
      >
        <div
          class="w-9/12 mx-auto"
        >
          <div
            class="relative h-16 sm:h-20 flex justify-center sm:justify-start"
          >
            <h2
              class="absolute mt-6 font-semibold text-blue-500"
            >
              CONTACT US
            </h2>
          </div>
          <p
            class="text-xs font-semibold text-blue-500"
          >
            Drop a message
          </p>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <input
              class="w-full mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            />
          </div>
          <textarea
            class="w-full h-40 mt-3 rounded-sm border border-gray-400 focus:outline-none focus:border-blue-500"
            placeholder="Message*"
          />
          <button
            class="w-full h-8 mt-4 text-xs rounded-full bg-blue-500 text-gray-200 tracking-wider focus:outline-none active:bg-blue-400 "
          >
            SUBMIT
          </button>
        </div>
      </div>
      <div
        class="absolute bottom-0 right-0 w-6/12 h-64 mx-auto hidden md:block"
      >
        <img
          alt="media1"
          class="h-full w-full object-contain"
          src="[object Object]"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`layouts Navbar Navbar User button: Navbar-layout-click-user-button 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-900"
  >
    <div
      class="fixed w-full h-10vh pt-4 text-gray-300 bg-gray-900 z-10 flex flex-row sm:justify-center sm:relative sm:w-9/12 sm:mx-auto"
      id="naver"
    >
      <div
        class="ml-2 sm:ml-0 text-xs md:text-sm lg:text-lg sm:absolute sm:left-0"
      >
        #
        <span
          class="text-purple-600"
        >
          Social
        </span>
         Media Manager
      </div>
      <button
        class="absolute right-0 top-0 mt-4 mx-4 z-10 sm:hidden border border-blue-500 text-blue-500 w-8 h-8 hover:bg-blue-500 hover:text-gray-200 focus:outline-none"
      >
        <i
          aria-hidden="true"
          class="fa fa-bars"
        />
      </button>
      <div
        class="sm:static sm:flex sm:pt-0"
        id="nav-container"
      >
        <nav
          class="text-gray-500 ml-2 sm:ml-0"
        >
          <ul
            class="text-xs md:text-base sm:grid sm:grid-flow-col sm:gap-8"
          >
            <li
              class="border-b border-gray-900 cursor-pointer rounded hover:bg-gray-800 sm:hover:bg-transparent sm:hover:border-green-500 sm:rounded-none"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="mt-1 sm:mt-0 border-b border-gray-900 cursor-pointer rounded hover:bg-gray-800 sm:hover:bg-transparent sm:hover:border-green-500 sm:rounded-none"
            >
              <a
                href="/group"
              >
                Media
              </a>
            </li>
            <li
              class="mt-1 sm:mt-0  border-b border-gray-900 cursor-pointer rounded hover:bg-gray-800 sm:hover:bg-transparent sm:hover:border-green-500 sm:rounded-none"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="w-12 pl-2 sm:pl-0 sm:absolute sm:right-0"
      >
        <button
          class="w-full border border-blue-500 bg-transparent text-blue-500 rounded-sm focus:outline-none hover:bg-blue-500 hover:text-gray-300"
          data-testid="user-button"
        >
          USER
        </button>
        <div
          class="w-48  mt-1 text-black bg-gray-100 flex flex-col rounded-md shadow-xl"
        >
          <a
            class="px-3 py-1 text-gray-800 hover:bg-indigo-500 hover:text-white"
            href="/"
          >
            Profile
          </a>
          <a
            class="px-3 py-1 text-gray-800 hover:bg-indigo-500 hover:text-white"
            href="/"
          >
            Settings
          </a>
          <a
            class="px-3 py-1 text-gray-800 hover:bg-indigo-500 hover:text-white"
            href="/controls"
          >
            Resources
          </a>
        </div>
      </div>
    </div>
    <div
      class="block sm:hidden w-full h-10vh"
    />
  </div>
</DocumentFragment>
`;

exports[`layouts Navbar Render Navbar: Navbar-layout 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-900"
  >
    <div
      class="fixed w-full h-10vh pt-4 text-gray-300 bg-gray-900 z-10 flex flex-row sm:justify-center sm:relative sm:w-9/12 sm:mx-auto"
      id="naver"
    >
      <div
        class="ml-2 sm:ml-0 text-xs md:text-sm lg:text-lg sm:absolute sm:left-0"
      >
        #
        <span
          class="text-purple-600"
        >
          Social
        </span>
         Media Manager
      </div>
      <button
        class="absolute right-0 top-0 mt-4 mx-4 z-10 sm:hidden border border-blue-500 text-blue-500 w-8 h-8 hover:bg-blue-500 hover:text-gray-200 focus:outline-none"
      >
        <i
          aria-hidden="true"
          class="fa fa-bars"
        />
      </button>
      <div
        class="sm:static sm:flex sm:pt-0"
        id="nav-container"
      >
        <nav
          class="text-gray-500 ml-2 sm:ml-0"
        >
          <ul
            class="text-xs md:text-base sm:grid sm:grid-flow-col sm:gap-8"
          >
            <li
              class="border-b border-gray-900 cursor-pointer rounded hover:bg-gray-800 sm:hover:bg-transparent sm:hover:border-green-500 sm:rounded-none"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="mt-1 sm:mt-0 border-b border-gray-900 cursor-pointer rounded hover:bg-gray-800 sm:hover:bg-transparent sm:hover:border-green-500 sm:rounded-none"
            >
              <a
                href="/group"
              >
                Media
              </a>
            </li>
            <li
              class="mt-1 sm:mt-0  border-b border-gray-900 cursor-pointer rounded hover:bg-gray-800 sm:hover:bg-transparent sm:hover:border-green-500 sm:rounded-none"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="w-12 pl-2 sm:pl-0 sm:absolute sm:right-0"
      >
        <button
          class="w-full border border-blue-500 bg-transparent text-blue-500 rounded-sm focus:outline-none hover:bg-blue-500 hover:text-gray-300"
          data-testid="user-button"
        >
          USER
        </button>
      </div>
    </div>
    <div
      class="block sm:hidden w-full h-10vh"
    />
  </div>
</DocumentFragment>
`;
